packages/core/src/components/button/_button.scss

```diff
@@ -260,6 +261,17 @@ a.pt-button {
   @extend .pt-button:disabled;
 }
 
+.pt-button-loading-spinner {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: absolute;
```

Comment:
My logic here (perhaps faulty) is to absolutely position the container to take the loading spinner out of flow, and then use flex so its child (the spinner) can be trivially positioned in the middle of this component.